packages/klighd-core/src/proxy-view/proxy-view-util.ts

```diff
@@ -0,0 +1,1001 @@
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { KGraphData } from "@kieler/klighd-interactive/lib/constraint-classes";
+import { injectable } from "inversify";
+import { VNode } from "snabbdom";
+import { ActionHandlerRegistry, IActionHandler, IActionHandlerInitializer, ICommand, SModelRoot } from "sprotty";
+import { Action, Bounds, isBounds, Dimension, Point, SelectAction, SelectAllAction, SetModelAction, UpdateModelAction, Viewport } from "sprotty-protocol";
+import { SendModelContextAction } from "../actions/actions";
+import { SKEdge, SKGraphElement, SKLabel, SKNode, SKPort } from "../skgraph-models";
+import { getElementByID } from "../skgraph-utils";
+
+//////// Interfaces ////////
+
+/**
+ * Contains all attributes used in defining a VNode's transform attribute.
+ * @example (x, y, width, height, scale, rotation)
+ */
+export interface TransformAttributes extends Bounds {
+    readonly scale?: number;
+    readonly rotation?: number;
+    readonly rotationPoint?: Point;
+}
+
+/** 
+ * Contains all canvas-related attributes.
+ * @acronym CRF - Canvas Reference Frame.
+ * @acronym LRF - Local Reference Frame.
+ * @example (x, y, width, height, scroll, zoom)
+ */
+export interface Canvas extends Viewport, Bounds {
+    /**
+     * Whether the canvas is in LRF, e.g. not in CRF.
+     * Usually doesn't need to be set explicitly - handled by translation methods.
+     * When the canvas hasn't been translated yet, this should be `undefined` or `false`,
+     * as the canvas should be in the CRF.
+     */
+    isInLRF?: boolean;
+}
+
+export namespace Canvas {
+    //// Get Canvas ////
+    /**
+     * Creates a canvas in CRF.
+     * @param boundsOrRoot The canvas' bounds or the root element.
+     * @param viewport The viewport.
+     * @returns The canvas.
+     */
+    export function of(boundsOrRoot: Bounds | SModelRoot, viewport: Viewport): Canvas {
+        const canvasBounds = isBounds(boundsOrRoot) ? boundsOrRoot : boundsOrRoot.canvasBounds;
+        return { ...canvasBounds, scroll: viewport.scroll, zoom: viewport.zoom };
+    }
+
+    //// Translation ////
+
+    /**
+     * Returns the bounds translated from the LRF to the CRF.
+     * E.g. calculates its position & width/height according to scroll and zoom.
+     * Inverse to {@link translateToLRF()}.
+     * @param bpd The bounds/point/dimension in the LRF.
+     * @param canvas The canvas.
+     * @returns The bounds translated to the CRF.
+     */
+    export function translateToCRF(bpd: Bounds | Point | Dimension, canvas: Canvas): Bounds {
+        const b = asBounds(bpd);
```

Comment:
const naming conventions and maybe more readable names

Reply:
the const is fine then, but i would still prefer the parameters to have better names. bpd for a parameter representing a bounds, point or dimension just doesn't really convey the intent